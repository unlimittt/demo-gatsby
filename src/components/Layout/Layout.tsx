import React, { ReactNode } from "react"
import styles from "./Layout.module.scss"
import SEO from "./SEO"

type Props = {
  title: string
  description?: string
  socialImage?: string
  children?: ReactNode
}

const Layout = ({ children, title, description, socialImage }: Props) => {

  return (
    <div className={styles.layout}>
      <SEO title={title} description={description} socialImage={socialImage} />
      {children}
    </div>
  )
}

export default Layout
