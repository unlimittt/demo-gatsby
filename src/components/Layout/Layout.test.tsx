import React from "react"
import siteMetadata from "../../../jest/__fixtures__/site-metadata"
import Layout from "./Layout"
import Enzyme, { shallow } from "enzyme"
import Adapter from "enzyme-adapter-react-16"

Enzyme.configure({ adapter: new Adapter() })

describe("Layout", () => {
  const props = {
    ...siteMetadata,
    description: "test",
    title: "test",
    children: "test",
  }

  it("renders correctly", () => {
    const wrapper = shallow(<Layout {...props} />)
    expect(wrapper).toMatchSnapshot()
  })
})
