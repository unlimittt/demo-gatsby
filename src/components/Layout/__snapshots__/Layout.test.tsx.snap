// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders correctly 1`] = `
<div
  className="layout"
>
  <SEO
    description="test"
    title="test"
  />
  test
</div>
`;
